### PR TITLE
Add support for __pow__ with pow() builtin

### LINKED ISSUE #4050: Inconsistency between ** and pow() with custom __pow__ implementations
When `__pow__` is overloaded in a custom class, the type checker is working as expected when using `**`, but generates a warning when using `pow()`.

POC:
```
class MyNum:
    def __init__(self, n: int):
        self.n = n
    
    def __pow__(self, power: "MyNum") -> "MyNum":
        return MyNum(pow(self.n, power.n))
    
    
x = MyNum(2)
y = MyNum(3)

a = x ** y  # works fine
b = pow(x, y)  # error

reveal_type(a)
reveal_type(b)
```
Output:
```
$ mypy pow_demo.py
pow_demo.py:14: error: No overload variant of "pow" matches argument types "MyNum", "MyNum"
pow_demo.py:14: note: Possible overload variants:
pow_demo.py:14: note:     def pow(base: int, exp: int, mod: None = ...) -> Any
pow_demo.py:14: note:     def pow(base: float, exp: float, mod: None = ...) -> float
pow_demo.py:14: note:     <1 more non-matching overload not shown>
pow_demo.py:16: note: Revealed type is 'pow_demo.MyNum'
pow_demo.py:17: note: Revealed type is 'Any'
```
Both a and b should be of the same type

### PR DESCRIPTION
**Pull request by markuspi**
Resolves #4050 

Adds typing support when the `pow()` builtin is used with instances of custom classes that implement the `__pow__` method.

Since the [documentation](https://docs.python.org/3/reference/datamodel.html#object.__pow__) states that it is optional to support the modulo parameter, I made sure that this choice is also reflected in the stub.

Below is a sample file that is compliant with these implemented changes. Only the disabled lines would generate errors, as expected.

```python
from typing import Optional

class Foo:
    pass

class MyNumA:  # Optional modulo
    def __pow__(self, exp: "MyNumA", mod: Optional[int] = None) -> Foo:
        return Foo()

class MyNumB:  # No modulo
    def __pow__(self, exp: "MyNumB") -> Foo:
        return Foo()

class MyNumC:  # Mandatory modulo
    def __pow__(self, exp: "MyNumC", mod: int) -> Foo:
        return Foo()

# ======
xa = MyNumA()
ya = MyNumA()

za1 = xa ** ya
za2 = pow(xa, ya)
za3 = pow(xa, ya, 2)

# ======
xb = MyNumB()
yb = MyNumB()

zb1 = xb ** yb
zb2 = pow(xb, yb)
#zb3 = pow(xb, yb, 2)  # raises error as expected

# ======
xc = MyNumC()
yc = MyNumC()

#zc1 = xc ** yc  # raises error as expected
#zc2 = pow(xc, yc)  # raises error as expected
zc3 = pow(xc, yc, 2)
```

### CODE REVIEW

File: stdlib/2/__builtin__.pyi [hunk 1]
```diff
@@ -1370,20 +1370,34 @@ else:
     # This is only available after from __future__ import print_function.
     def print(*values: object, sep: Optional[Text] = ..., end: Optional[Text] = ..., file: Optional[_Writer] = ...) -> None: ...
 
+_E = TypeVar("_E", contravariant=True)
```
Comment (JelleZijlstra) [inline stdlib/2/__builtin__.pyi:1373]: I'm having a hard time understanding the need for contravariance here. I tested with the following file:

```
class A:
    pass


class B(A):
    pass


class C(B):
    pass


class PowPow:
    def __pow__(self, exp: B) -> str:
        return ""


reveal_type(pow(PowPow(), A()))
reveal_type(pow(PowPow(), B()))
reveal_type(pow(PowPow(), C()))
```

But mypy produces the same output regardless of whether I make `_E` contravariant, covariant, or invariant:
```
powpow.py:18: error: Cannot infer type argument 1 of "pow"
powpow.py:18: note: Revealed type is 'Any'
powpow.py:19: note: Revealed type is 'builtins.str*'
powpow.py:20: note: Revealed type is 'builtins.str*'
```

Comment (markuspi) [reply stdlib/2/__builtin__.pyi:1373]: I wasn't sure about the variance. Looks like your're right. Is my `_T_co` needed for the return value?

Comment (markuspi) [reply stdlib/2/__builtin__.pyi:1373]: `test/mypy_test.py` reports an error when `_E` and `_M` are not contravariant:
```
$ python3 tests/mypy_test.py
running mypy --python-version 3.9 --strict-optional --no-site-packages --show-traceback --no-implicit-optional --disallow-any-generics --disallow-subclassing-any # with 970 files
stdlib/2and3/builtins.pyi:1375: error: Invariant type variable '_E' used in protocol where contravariant one is expected
stdlib/2and3/builtins.pyi:1377: error: Invariant type variable '_E' used in protocol where contravariant one is expected
stdlib/2and3/builtins.pyi:1377: error: Invariant type variable '_M' used in protocol where contravariant one is expected
```

Comment (JelleZijlstra) [reply stdlib/2/__builtin__.pyi:1373]: OK, I guess we should believe mypy here.

Comment (hauntsaninja) [reply stdlib/2/__builtin__.pyi:1373]: mypy's a little strict about variance in generic protocols. Say you subclassed PowPow with PowPow2 that accepted a wider variety of types than just B's. PowPow2 does obey LSP, and is a nominal subtype of PowPow, but is not a subtype of Pow[B] if the type variables are invariant, resulting in weird intransitivity.

**Pull request merged**